### ISSUE TITLE
Cannot use Kernel Debugger together with VMWare video driver 

### ISSUE THREAD

**Issue by volo-droid** [HaikuArchives/VMwareAddons#32]
After entering the Kernel Debugger (either manually or after a kernel panic) the screen freezes, if Haiku is using VMWare video driver. Blindly typing `exit` on the keyboard (followed by the Enter keypress) brings us back to the OS, and everything works as before. Is it an expected behaviour, and if so, is there any workaround to it?

**Comment by diversys** [comment]
This is a bug in vmware graphics driver which is not part of Haiku. That's why it was moved to #9 in the first place.

**Comment by volo-droid** [comment]
I see your point. But is it possible to fix the issue in the VmWare driver itself without modifying Haiku's KDL internals?
There were a few suggestions in the ticket, which led me to think it could be fixed only by altering `kernel/debug/…` parts:
 
> No, KDL runs with interrupts off, and that also includes timer interrupts. Is it possible to let VMware update the screen automatically again somehow (like in VESA mode)? Another possibility would be to alter kernel/debug/frame_buffer_console.cpp to tell VMware when to update the screen. A cleaner solution to this would be to have a special frame buffer module and have kernel/debug/blue_screen.cpp use that one instead, when the VMware driver is running.

**Comment by diversys** [comment]
Maybe but vmware_addons are in maintenance mode. If you know how to fix it please do so and I'd be happy to merge your fix :)

**Comment by volo-droid** [comment]
Nah, those changes will fix the build process, but the resulting driver will be unusable. I made a similar mistake yesterday 😄 
I'll publish the solution I've come too asap, just let me cleanup my code a little bit.

**Comment by diversys** [comment]
Cool, thanks! :)
